scripts/get-flattened-publication.sh

```diff
@@ -60,17 +60,26 @@ curl -sS "$URL_PREFIX" --request POST --header 'Content-Type: application/json'
 )" | \
 # Decode, simplify, flatten, and compact the output.
 jq -c '
-  # Capture block height.
-  .result.response.height? as $height |
+  # Capture response block height.
+  .result.response.height? as $responseHeight |
 
   # Decode `value` as base64, then decode that as JSON.
   .result.response.value | @base64d | fromjson |
 
-  # Decode `value` as JSON, capture `slots`, then decode `body` as JSON.
-  .value | fromjson | .slots as $slots | .body | fromjson |
+  # Decode `value` as JSON.
+  .value | fromjson |
 
-  # Add block height.
-  (.blockHeight |= $height) |
+  # Upgrade a naked value to a stream cell if necessary.
+  if has("height") and has("values") then . else { values: [ . | tojson ] } end |
+
+  # Capture data block height.
+  .height as $dataHeight |
+
+  # Flatten each value independently.
+  .values[] | fromjson |
```

Comment:
I wondered if this change hand any impact on this script. Do we have any tests that would have failed if we didn't remember to update this script?

Reply:
I'm pretty sure we don't have any tests of shell scripts, and I think that's fine. They're not exported from the packages and not part of the API of the package. They're just here to help organize the code.

Reply:
It's perhaps not exported in the npm sense, but it's a customer-visible API (customer-required, in fact). I would like to see an integration test where
  - javascript code writes capData to a file
  - this script reads the capData file and writes flat data
  - we diff the flat data against expected results

Of course, I'd like to see a lot of things... I'm not sure whether it's cost-effective any time soon. It's a question of how often we're likely to tweak this script.

Reply:
Perhaps it _should_ be exported in the npm sense? that is: perhaps it should go in `.bin`?

Reply:
The problem with a test like you describe above is that it's mostly navel-gazing—there's no connection between the hard-coded input and the kind of output that is expected of RPC nodes in the real world. But I do still thank that having _something_ is a good idea.

Reply:
OK, I've added shell-based testing. Any ideas for how to make sure they run with the rest of our tests?

Reply:
The `test` nom script could call the shell test thing, maybe? (I haven't looked at the shell thing)

Reply:
Added a simple runner at [scripts/test/test.sh](https://github.com/Agoric/agoric-sdk/pull/5942/files#diff-686b1343c78734cb242be3e7a388cfe4cbc4c80d6e071327c6b14446518eff2c) (and a [GitHub workflow calling it](https://github.com/Agoric/agoric-sdk/pull/5942/files#diff-efee01ea8ffefa44cbe15acfb318800ac5cceef28e30c1452cb8f84b6070e49f)).